src/main/java/org/jenkinsci/plugins/displayurlapi/ClassicDisplayURLProvider.java

```diff
@@ -4,6 +4,7 @@
 import hudson.Util;
 import hudson.model.Job;
 import hudson.model.Run;
+import javax.annotation.Nonnull;
```

Comment:
Where is the JSR release defining this `javax` namespaced annotation? It is very dangerous to use annotations in `javax` unless they have been *published* through the JSR process. I would advise against using this annotation unless you can point to an official JSR release, especially given that the java module system does not allow two modules to share the same package names and IIRC an existing module is already publishing the `javax.annotation` package name (the module is `java.xml.ws.annotation`) so using this annotation would also cause issues with migrating Jenkins to support the module path.

In short, unless and until there is a JSR that *publishes* this annotation it is extremely limiting on the evolution of Jenkins to continue using it. Use the one from spotbugs instead: https://static.javadoc.io/com.github.spotbugs/spotbugs-annotations/3.1.8/edu/umd/cs/findbugs/annotations/NonNull.html

Reply:
it was used previously in the package so I thought, what the heck but I totally agree 🤭